packages/core/src/html/headerProcessor.ts

```diff
@@ -12,22 +12,26 @@ const _ = {
 /*
  * h1 - h6
  */
-export function setHeadingId(node: MbNode, config: NodeProcessorConfig) {
+export function setHeadingId(node: MbNode,
+                             config: NodeProcessorConfig,
+                             callFromSiteLinkManager: boolean = false) {
   const textContent = cheerio(node).text();
   // remove the '&lt;' and '&gt;' symbols that markdown-it uses to escape '<' and '>'
   const cleanedContent = textContent.replace(/&lt;|&gt;/g, '');
   const slugifiedHeading = slugify(cleanedContent, { decamelize: false });
 
   let headerId = slugifiedHeading;
   const { headerIdMap } = config;
-  if (headerIdMap[slugifiedHeading]) {
-    headerId = `${slugifiedHeading}-${headerIdMap[slugifiedHeading]}`;
-    headerIdMap[slugifiedHeading] += 1;
-  } else {
-    headerIdMap[slugifiedHeading] = 2;
+  if (!callFromSiteLinkManager) {
+    if (headerIdMap[slugifiedHeading]) {
+      headerId = `${slugifiedHeading}-${headerIdMap[slugifiedHeading]}`;
+      headerIdMap[slugifiedHeading] += 1;
+    } else {
+      headerIdMap[slugifiedHeading] = 2;
+    }
+    node.attribs.id = headerId;
```

Comment:
Why is the headerId not set when called by SiteLinkManager?

Reply:
This is to prevent regression.

<img width="1064" alt="Screenshot 2024-03-25 at 01 02 07" src="https://github.com/MarkBind/markbind/assets/121547057/2d0adb79-9e7e-418d-9b6f-0735c1ac0835">
<img width="755" alt="Screenshot 2024-03-25 at 01 01 56" src="https://github.com/MarkBind/markbind/assets/121547057/8e76c8d1-3a86-444e-af3a-4557b495e78d">

The validation is only for check purpose, and should not modify the current state;

Reply:
Regarding the aspect not related to internal state: 

When the "maintainHashesForInclude" is called, the heading names is already processed, and we do not know how many same headings has been processed since then, so we can not just anyhow append the name with the number of times it has occured (or minus something).

The appending with the number of times something occur is because id must by unique. This happens internally and the user does not know, so we should not be expecting user to include the -${counter} in the first place.

A not so important but still valid reason is that, the existence of ${name}-{counter} necessitates the existence of ${name}, so no matter what the current state is, it is never wrong to return the name without counter.